src/lorarmesh.cpp

```diff
@@ -14,29 +30,64 @@ LoraMesh::LoraMesh(DataQueue<message_t>* txQueue,
   this->txQueue = txQueue;
   this->rxQueue = rxQueue;
   this->router = router;
+
+  Layer1 = new Layer1Class();
+  Layer1->setPins(LORA_CS, LORA_RST, LORA_IRQ);
+  Layer1->setTxPower(TX_POWER);
+  Layer1->setLoRaFrequency(LORA_FREQ);
+
+  int attempts = MAX_BOOT_LL1_RETRIES;
+
+  while (!Layer1->init()) {
+    delay(BOOT_LL1_DELAY_MICROS);
+    if (attempts-- == 0)
+      ESP.restart();
+  }
+  LL2 = new LL2Class(Layer1);
+  LL2->setInterval(10000);  // this value needs more research
+  LL2->init();
+
   xTaskCreatePinnedToCore(task, "LoraMeshTransceiver", 10000, this, 0,
                           &transceiverTaskHandle, LORA_TASKS_CORE);
 }
 
 void LoraMesh::transmit() {
   auto message = txQueue->poll();
+
   if (message == nullptr)
     return;
 
+  LL2->daemon();
+  struct Datagram datagram;
+  int datagramsize = DATAGRAM_HEADER;
+  datagram.type = message->type;
+  bool checkTransmitMessage = false;
+
   if (message->type == CONTROL_MESSAGE) {
-    // TODO broadcast message over LoRaLayer2
+    memcpy(datagram.destination, BROADCAST_NODES, ADDR_LENGTH);
+    memcpy(datagram.message, &message->data.control, sizeof(control_data_t));
+    datagramsize += sizeof(control_data_t);
+    checkTransmitMessage = true;
+
   } else if (message->type == DATA_MESSAGE) {
     auto destinationAddr = router->getGatewayAddress();
-    if (destinationAddr == 0)
-      goto freeData;
+    if (destinationAddr > 0) {
+      memcpy(datagram.destination, &destinationAddr, ADDR_LENGTH);
+      memcpy(datagram.message, message->data.layer2.payload,
+             message->data.layer2.length);
+      datagramsize += message->data.layer2.length;
+      checkTransmitMessage = true;
+    }
+  }
 
-    // TODO send to destinationAddr through LoRaLayer2
+  if (checkTransmitMessage) {
```

Comment:
[OPTIONAL] Rename this variable to "shouldTransmitt" or check `if (datagramsize > DATAGRAM_HEADER)` so you don't neeed a separate flag...

Reply:
We rename the variable to shouldtransmit